common/src/main/java/com/wynntils/core/managers/ManagerRegistry.java

```diff
@@ -59,81 +59,79 @@ public static void init() {
+            throw new IllegalStateException("Tried to unregister a core manager like a model.");
         }
 
-        MODEL_DEPENDENCIES.putIfAbsent(dependency, new ArrayList<>());
+        List<ModelDependant> modelDependencies = MODEL_DEPENDENCIES.get(dependency);
 
-        MODEL_DEPENDENCIES.get(dependency).remove(dependant);
+        // Check if present and try to remove
+        if (modelDependencies == null || !modelDependencies.remove(dependant)) {
+            WynntilsMod.warn(
+                    String.format("Could not remove dependency of %s for %s when lacking", dependant, dependency));
+            return;
+        }
 
-        updateModelState(dependency);
-    }
+        // Check if should disable manager
+        if (modelDependencies.isEmpty()) {
+            // remove empty list
+            MODEL_DEPENDENCIES.remove(dependency);
 
-    public static void removeAllDependencies(ModelDependant dependant) {
-        for (Class<? extends Model> model : dependant.getModelDependencies()) {
-            if (MODEL_DEPENDENCIES.containsKey(model)
-                    && MODEL_DEPENDENCIES.get(model).remove(dependant)) {
-                updateModelState(model);
-            }
+            ENABLED_MANAGERS.remove(dependency);
+            tryDisableManager(dependency);
         }
     }
 
-    private static void updateModelState(Class<? extends Model> model) {
-        List<ModelDependant> dependencies = MODEL_DEPENDENCIES.get(model);
-
-        boolean hasDependencies = dependencies != null && !dependencies.isEmpty();
-
-        if (ENABLED_MANAGERS.contains(model)) {
-            if (!hasDependencies) {
-                WynntilsMod.unregisterEventListener(model);
-
-                ENABLED_MANAGERS.remove(model);
-
-                tryDisableManager(model);
-            }
-        } else {
-            if (hasDependencies) {
-                WynntilsMod.registerEventListener(model);
-
-                ENABLED_MANAGERS.add(model);
+    public static void addAllDependencies(ModelDependant dependant) {
+        for (Class<? extends Model> dependency : dependant.getModelDependencies()) {
+            addDependency(dependant, dependency);
+        }
+    }
 
-                tryInitManager(model);
-            }
+    public static void removeAllDependencies(ModelDependant dependant) {
+        for (Class<? extends Model> dependency : dependant.getModelDependencies()) {
+            removeDependency(dependant, dependency);
         }
     }
 
     private static void tryInitManager(Class<? extends Manager> manager) {
+        WynntilsMod.registerEventListener(manager);
+
         try {
             MethodUtils.invokeExactStaticMethod(manager, "init");
         } catch (IllegalAccessException | NoSuchMethodException e) {
-            WynntilsMod.warn("Misconfigured init() on manager " + manager, e);
-            assert false;
+            WynntilsMod.error("Misconfigured init() on manager " + manager, e);
```

Comment:
Throw a runtime exception instead. We need to catch this design errors during development.

I'll review the rest of the PR later.

Reply:
Th other branch of the try-catch does not do that, should this also be changed?

Reply:
Look at the exceptions, and think of what causes them to be thrown, and why they differ. What can possibly cause any of the two first exceptions? And what can cause the last?

Reply:
Technically the second is more of a "compile-time" problem

Reply:
Anyway, I won't have time to do much more reviewing of all this refactoring PRs until Fridays at earliest, sorry 'bout that.

Reply:
Please either reinstate the assert, or change this into throwing a RuntimeException. Otherwise it is a regression in our ability to find structural code errors during development.